haybaler_taxonomy.py

```diff
@@ -17,11 +20,44 @@ def read_csv(file, path):
     return csv
 
 
+# work in progress - not used yet
+def find_species(csv):
+    species = []
+    for organism in csv.index:
+        organism = organism.replace("organism_", "")
+        organism = organism.replace("1_1_1", "Homo_sapiens_")
+        organism = re.sub(r'^.*?:', '', organism)
+        m = re.findall('[A-Z][a-z]+_[a-z]+', organism)
+        if len(m) != 0:
+            m[0] = m[0].replace("_", " ")
+            name = (m[0])
+        else:
+            name = "NOT KNOWN"
+        print(name)
+        species.append(name)
+    taxonomy = pytaxonkit.name2taxid(species)
+    nan_list = set(taxonomy[taxonomy["TaxID"].isna()]["Name"].to_list())  # list of names that produce NAN
+    genus_series = pd.Series(species)
+    csv.insert(loc=0, column='genus', value=genus_series.values)
+    print(csv[csv['genus'].isin(nan_list)]["genus"])  # print everything that didn't work with pytaxonkit
+    total_chr = len(species)
+    chr_not_work = len(taxonomy[taxonomy["TaxID"].isna()])
+    chr_work = total_chr - chr_not_work
```

Comment:
#TODO -  this is not neccessarily the case (edge case problems!) Better to count chr_work and well as chr_not_work

Reply:
The Problem there are again the doubble assigned TaxIDs. Counting what not works is easy, if it works not onbviously no TaxID is assigned so they cant be doubble. But we still can't find these doubble Taxa. 
Of course in the current script at the end a dataframe is saved and printed with all genuses or  a "NOT KNOWN" and I could count everything that is not "NOT KNOWN". But I just replace everything that didn't work with "NOT KNOWN" so the calculation would be the same: "everything minus what not worked". I could write a function to find these doubble Taxa but that would probably return more mistakes than this method...

Reply:
I see ! OK, thanks for the explanation. We can leave it as is then, I just thought it might lead to errors.